beacon_node/src/config.rs

```diff
@@ -1357,79 +1358,78 @@ pub fn set_network_config(
 
-    if cli_args.is_present("private") {
+    if parse_flag(cli_args, "private") {
         config.private = true;
     }
 
-    if cli_args.is_present("metrics") {
+    if parse_flag(cli_args, "metrics") {
         config.metrics_enabled = true;
     }
 
-    if cli_args.is_present("enable-private-discovery") {
+    if parse_flag(cli_args, "enable-private-discovery") {
         config.discv5_config.table_filter = |_| true;
     }
 
     // Light client server config.
-    config.enable_light_client_server = cli_args.is_present("light-client-server");
-
-    // The self limiter is disabled by default.
-    // This flag can be used both with or without a value. Try to parse it first with a value, if
-    // no value is defined but the flag is present, use the default params.
-    config.outbound_rate_limiter_config = clap_utils::parse_optional(cli_args, "self-limiter")?;
-    if cli_args.is_present("self-limiter") && config.outbound_rate_limiter_config.is_none() {
-        config.outbound_rate_limiter_config = Some(Default::default());
+    config.enable_light_client_server = parse_flag(cli_args, "light-client-server");
+
+    // The self limiter is disabled by default. If the `self-limiter` flag is provided
+    // without the `self-limiter-protocols` flag, the default params will be used.
+    if parse_flag(cli_args, "self-limiter") {
+        config.outbound_rate_limiter_config =
+            if let Some(protocols) = cli_args.get_one::<String>("self-limiter-protocols") {
+                Some(protocols.parse()?)
+            } else {
+                Some(Default::default())
+            };
     }
 
     // Proposer-only mode overrides a number of previous configuration parameters.
     // Specifically, we avoid subscribing to long-lived subnets and wish to maintain a minimal set
     // of peers.
-    if cli_args.is_present("proposer-only") {
+    if parse_flag(cli_args, "proposer-only") {
         config.subscribe_all_subnets = false;
 
-        if cli_args.value_of("target-peers").is_none() {
+        if cli_args.get_one::<String>("target-peers").is_none() {
             // If a custom value is not set, change the default to 15
             config.target_peers = 15;
         }
         config.proposer_only = true;
         warn!(log, "Proposer-only mode enabled"; "info"=> "Do not connect a validator client to this node unless via the --proposer-nodes flag");
     }
-    // The inbound rate limiter is enabled by default unless `disabled` is passed to the
-    // `inbound-rate-limiter` flag. Any other value should be parsed as a configuration string.
-    config.inbound_rate_limiter_config = match cli_args.value_of("inbound-rate-limiter") {
-        None => {
-            // Enabled by default, with default values
+    // The inbound rate limiter is enabled by default unless `disabled` via the
+    // `disable-inbound-rate-limiter` flag.
+    config.inbound_rate_limiter_config = if parse_flag(cli_args, "disable-inbound-rate-limiter") {
```

Comment:
Changing from `inbound-rate-limiter` to `disable-inbound-rate-limiter` appears to be a breaking change on the beacon subcommand, is it necessary?

AFAIU the current flag usage is to disable is `--inbound-rate-limiter=disable`?

Reply:
Currently this cli param is "bad" because it tries to do too many things. By default its value is set to `true`. if the cli param is provided without a value, its treated as `true`. On top of this the param can either accept the value `disable` OR a `;` separated list of protocol values w/ the following format: `<protocol_name>:<tokens>/<time_in_seconds>`

Technically these breaking changes aren't necessary, however the new version of clap is a bit stricter with cli param definitions. We now either define the cli param as accepting a value (`ArgAction::Set`) or as a bool flag (`ArgAction::SetTrue`). The existing functionality requires us to treat this cli param as a hybrid of `Set` and `SetTrue` like so

```
.default_missing_value("true")
.require_equals(true)
.num_args(0..1)
```

which is also technically a breaking change since the cli param now requires an `=` sign if the user tries to provide it a value (currently the `=` isn't necessary). I think its a bit nicer to just keep all cli params as either `Set` or `SetTrue`, but I'm open to the hybrid option if we decide this breaking change isnt worth it.

Reply:
If @AgeManning blesses this breaking change I'm okay with it

Reply:
Yeah I'm down for breaking changes if it conforms better to the new clap. Makes sense to me.